// pdfstamp/dist/utils/signature-utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalculateOrientation = exports.CalculateZoom = void 0;
function CalculateZoom(userZoom, pageWidth, imageWidth) {
    var imageBiggerFactor = imageWidth / pageWidth;
    var realZoom = userZoom / imageBiggerFactor;
    return realZoom;
}
exports.CalculateZoom = CalculateZoom;
function CalculateOrientation(isUsingBottom, isUsingLeft, moveL, moveR, moveT, moveB) {
    var isUsingTop = !isUsingBottom;
    var isUsingRight = !isUsingLeft;
    if (isUsingTop && isUsingLeft) {
        return {
            isUsingTop: isUsingTop,
            isUsingLeft: isUsingLeft,
            gravity: 'northwest',
            y: moveT,
            x: moveL,
        };
    }
    if (isUsingTop && isUsingRight) {
        return {
            isUsingTop: isUsingTop,
            isUsingLeft: isUsingLeft,
            gravity: 'northeast',
            y: moveT,
            x: moveR,
        };
    }
    if (isUsingBottom && isUsingLeft) {
        return {
            isUsingTop: isUsingTop,
            isUsingLeft: isUsingLeft,
            gravity: 'southwest',
            y: moveB,
            x: moveL,
        };
    }
    // if (isUsingBottom && isUsingRight) {
    return {
        isUsingTop: isUsingTop,
        isUsingLeft: isUsingLeft,
        gravity: 'southeast',
        y: moveB,
        x: moveR,
    };
}
exports.CalculateOrientation = CalculateOrientation;
