// mslpsapi/src/model/AcroFields.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/BaseFont', 'model/Item', 'model/TextField', 'model/XfaForm'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./BaseFont'), require('./Item'), require('./TextField'), require('./XfaForm'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.AcroFields = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.BaseFont, root.Mslpsapi.Item, root.Mslpsapi.TextField, root.Mslpsapi.XfaForm);
  }
}(this, function(ApiClient, BaseFont, Item, TextField, XfaForm) {
  'use strict';




  /**
   * The AcroFields model module.
   * @module model/AcroFields
   * @version v1
   */

  /**
   * Constructs a new <code>AcroFields</code>.
   * @alias module:model/AcroFields
   * @class
   */
  var exports = function() {
    var _this = this;







  };

  /**
   * Constructs a <code>AcroFields</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/AcroFields} obj Optional instance to populate.
   * @return {module:model/AcroFields} The populated <code>AcroFields</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('Fields')) {
        obj['Fields'] = ApiClient.convertToType(data['Fields'], {'String': Item});
      }
      if (data.hasOwnProperty('GenerateAppearances')) {
        obj['GenerateAppearances'] = ApiClient.convertToType(data['GenerateAppearances'], 'Boolean');
      }
      if (data.hasOwnProperty('TotalRevisions')) {
        obj['TotalRevisions'] = ApiClient.convertToType(data['TotalRevisions'], 'Integer');
      }
      if (data.hasOwnProperty('FieldCache')) {
        obj['FieldCache'] = ApiClient.convertToType(data['FieldCache'], {'String': TextField});
      }
      if (data.hasOwnProperty('SubstitutionFonts')) {
        obj['SubstitutionFonts'] = ApiClient.convertToType(data['SubstitutionFonts'], [BaseFont]);
      }
      if (data.hasOwnProperty('Xfa')) {
        obj['Xfa'] = XfaForm.constructFromObject(data['Xfa']);
      }
    }
    return obj;
  }

  /**
   * @member {Object.<String, module:model/Item>} Fields
   */
  exports.prototype['Fields'] = undefined;
  /**
   * @member {Boolean} GenerateAppearances
   */
  exports.prototype['GenerateAppearances'] = undefined;
  /**
   * @member {Integer} TotalRevisions
   */
  exports.prototype['TotalRevisions'] = undefined;
  /**
   * @member {Object.<String, module:model/TextField>} FieldCache
   */
  exports.prototype['FieldCache'] = undefined;
  /**
   * @member {Array.<module:model/BaseFont>} SubstitutionFonts
   */
  exports.prototype['SubstitutionFonts'] = undefined;
  /**
   * @member {module:model/XfaForm} Xfa
   */
  exports.prototype['Xfa'] = undefined;



  return exports;
}));
