// mslpsapi/src/model/Address.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.Address = factory(root.Mslpsapi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The Address model module.
   * @module model/Address
   * @version v1
   */

  /**
   * Constructs a new <code>Address</code>.
   * @alias module:model/Address
   * @class
   */
  var exports = function() {
    var _this = this;
















  };

  /**
   * Constructs a <code>Address</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Address} obj Optional instance to populate.
   * @return {module:model/Address} The populated <code>Address</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('Country')) {
        obj['Country'] = ApiClient.convertToType(data['Country'], 'String');
      }
      if (data.hasOwnProperty('CountryName')) {
        obj['CountryName'] = ApiClient.convertToType(data['CountryName'], 'String');
      }
      if (data.hasOwnProperty('Province')) {
        obj['Province'] = ApiClient.convertToType(data['Province'], 'String');
      }
      if (data.hasOwnProperty('City')) {
        obj['City'] = ApiClient.convertToType(data['City'], 'String');
      }
      if (data.hasOwnProperty('Postcode')) {
        obj['Postcode'] = ApiClient.convertToType(data['Postcode'], 'String');
      }
      if (data.hasOwnProperty('StreetAddress')) {
        obj['StreetAddress'] = ApiClient.convertToType(data['StreetAddress'], 'String');
      }
      if (data.hasOwnProperty('Name')) {
        obj['Name'] = ApiClient.convertToType(data['Name'], 'String');
      }
      if (data.hasOwnProperty('Phone')) {
        obj['Phone'] = ApiClient.convertToType(data['Phone'], 'String');
      }
      if (data.hasOwnProperty('Mobile')) {
        obj['Mobile'] = ApiClient.convertToType(data['Mobile'], 'String');
      }
      if (data.hasOwnProperty('Email')) {
        obj['Email'] = ApiClient.convertToType(data['Email'], 'String');
      }
      if (data.hasOwnProperty('TrademanageId')) {
        obj['TrademanageId'] = ApiClient.convertToType(data['TrademanageId'], 'String');
      }
      if (data.hasOwnProperty('AddressId')) {
        obj['AddressId'] = ApiClient.convertToType(data['AddressId'], 'String');
      }
      if (data.hasOwnProperty('Fax')) {
        obj['Fax'] = ApiClient.convertToType(data['Fax'], 'String');
      }
      if (data.hasOwnProperty('County')) {
        obj['County'] = ApiClient.convertToType(data['County'], 'String');
      }
      if (data.hasOwnProperty('Company')) {
        obj['Company'] = ApiClient.convertToType(data['Company'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} Country
   */
  exports.prototype['Country'] = undefined;
  /**
   * @member {String} CountryName
   */
  exports.prototype['CountryName'] = undefined;
  /**
   * @member {String} Province
   */
  exports.prototype['Province'] = undefined;
  /**
   * @member {String} City
   */
  exports.prototype['City'] = undefined;
  /**
   * @member {String} Postcode
   */
  exports.prototype['Postcode'] = undefined;
  /**
   * @member {String} StreetAddress
   */
  exports.prototype['StreetAddress'] = undefined;
  /**
   * @member {String} Name
   */
  exports.prototype['Name'] = undefined;
  /**
   * @member {String} Phone
   */
  exports.prototype['Phone'] = undefined;
  /**
   * @member {String} Mobile
   */
  exports.prototype['Mobile'] = undefined;
  /**
   * @member {String} Email
   */
  exports.prototype['Email'] = undefined;
  /**
   * @member {String} TrademanageId
   */
  exports.prototype['TrademanageId'] = undefined;
  /**
   * @member {String} AddressId
   */
  exports.prototype['AddressId'] = undefined;
  /**
   * @member {String} Fax
   */
  exports.prototype['Fax'] = undefined;
  /**
   * @member {String} County
   */
  exports.prototype['County'] = undefined;
  /**
   * @member {String} Company
   */
  exports.prototype['Company'] = undefined;



  return exports;
}));
