// mslpsapi/src/model/BaseColor.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.BaseColor = factory(root.Mslpsapi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The BaseColor model module.
   * @module model/BaseColor
   * @version v1
   */

  /**
   * Constructs a new <code>BaseColor</code>.
   * @alias module:model/BaseColor
   * @class
   */
  var exports = function() {
    var _this = this;






  };

  /**
   * Constructs a <code>BaseColor</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/BaseColor} obj Optional instance to populate.
   * @return {module:model/BaseColor} The populated <code>BaseColor</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('RGB')) {
        obj['RGB'] = ApiClient.convertToType(data['RGB'], 'Integer');
      }
      if (data.hasOwnProperty('R')) {
        obj['R'] = ApiClient.convertToType(data['R'], 'Integer');
      }
      if (data.hasOwnProperty('G')) {
        obj['G'] = ApiClient.convertToType(data['G'], 'Integer');
      }
      if (data.hasOwnProperty('B')) {
        obj['B'] = ApiClient.convertToType(data['B'], 'Integer');
      }
      if (data.hasOwnProperty('A')) {
        obj['A'] = ApiClient.convertToType(data['A'], 'Integer');
      }
    }
    return obj;
  }

  /**
   * @member {Integer} RGB
   */
  exports.prototype['RGB'] = undefined;
  /**
   * @member {Integer} R
   */
  exports.prototype['R'] = undefined;
  /**
   * @member {Integer} G
   */
  exports.prototype['G'] = undefined;
  /**
   * @member {Integer} B
   */
  exports.prototype['B'] = undefined;
  /**
   * @member {Integer} A
   */
  exports.prototype['A'] = undefined;



  return exports;
}));
