// mslpsapi/src/model/ByteBuffer.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.ByteBuffer = factory(root.Mslpsapi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The ByteBuffer model module.
   * @module model/ByteBuffer
   * @version v1
   */

  /**
   * Constructs a new <code>ByteBuffer</code>.
   * @alias module:model/ByteBuffer
   * @class
   */
  var exports = function() {
    var _this = this;











  };

  /**
   * Constructs a <code>ByteBuffer</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ByteBuffer} obj Optional instance to populate.
   * @return {module:model/ByteBuffer} The populated <code>ByteBuffer</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('Size')) {
        obj['Size'] = ApiClient.convertToType(data['Size'], 'Integer');
      }
      if (data.hasOwnProperty('Buffer')) {
        obj['Buffer'] = ApiClient.convertToType(data['Buffer'], 'String');
      }
      if (data.hasOwnProperty('CanRead')) {
        obj['CanRead'] = ApiClient.convertToType(data['CanRead'], 'Boolean');
      }
      if (data.hasOwnProperty('CanSeek')) {
        obj['CanSeek'] = ApiClient.convertToType(data['CanSeek'], 'Boolean');
      }
      if (data.hasOwnProperty('CanWrite')) {
        obj['CanWrite'] = ApiClient.convertToType(data['CanWrite'], 'Boolean');
      }
      if (data.hasOwnProperty('Length')) {
        obj['Length'] = ApiClient.convertToType(data['Length'], 'Integer');
      }
      if (data.hasOwnProperty('Position')) {
        obj['Position'] = ApiClient.convertToType(data['Position'], 'Integer');
      }
      if (data.hasOwnProperty('CanTimeout')) {
        obj['CanTimeout'] = ApiClient.convertToType(data['CanTimeout'], 'Boolean');
      }
      if (data.hasOwnProperty('ReadTimeout')) {
        obj['ReadTimeout'] = ApiClient.convertToType(data['ReadTimeout'], 'Integer');
      }
      if (data.hasOwnProperty('WriteTimeout')) {
        obj['WriteTimeout'] = ApiClient.convertToType(data['WriteTimeout'], 'Integer');
      }
    }
    return obj;
  }

  /**
   * @member {Integer} Size
   */
  exports.prototype['Size'] = undefined;
  /**
   * @member {String} Buffer
   */
  exports.prototype['Buffer'] = undefined;
  /**
   * @member {Boolean} CanRead
   */
  exports.prototype['CanRead'] = undefined;
  /**
   * @member {Boolean} CanSeek
   */
  exports.prototype['CanSeek'] = undefined;
  /**
   * @member {Boolean} CanWrite
   */
  exports.prototype['CanWrite'] = undefined;
  /**
   * @member {Integer} Length
   */
  exports.prototype['Length'] = undefined;
  /**
   * @member {Integer} Position
   */
  exports.prototype['Position'] = undefined;
  /**
   * @member {Boolean} CanTimeout
   */
  exports.prototype['CanTimeout'] = undefined;
  /**
   * @member {Integer} ReadTimeout
   */
  exports.prototype['ReadTimeout'] = undefined;
  /**
   * @member {Integer} WriteTimeout
   */
  exports.prototype['WriteTimeout'] = undefined;



  return exports;
}));
