// mslpsapi/src/model/FieldInformation.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/PRIndirectReference', 'model/PdfDictionary'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./PRIndirectReference'), require('./PdfDictionary'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.FieldInformation = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.PRIndirectReference, root.Mslpsapi.PdfDictionary);
  }
}(this, function(ApiClient, PRIndirectReference, PdfDictionary) {
  'use strict';




  /**
   * The FieldInformation model module.
   * @module model/FieldInformation
   * @version v1
   */

  /**
   * Constructs a new <code>FieldInformation</code>.
   * @alias module:model/FieldInformation
   * @class
   */
  var exports = function() {
    var _this = this;





  };

  /**
   * Constructs a <code>FieldInformation</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/FieldInformation} obj Optional instance to populate.
   * @return {module:model/FieldInformation} The populated <code>FieldInformation</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('WidgetName')) {
        obj['WidgetName'] = ApiClient.convertToType(data['WidgetName'], 'String');
      }
      if (data.hasOwnProperty('Name')) {
        obj['Name'] = ApiClient.convertToType(data['Name'], 'String');
      }
      if (data.hasOwnProperty('Info')) {
        obj['Info'] = PdfDictionary.constructFromObject(data['Info']);
      }
      if (data.hasOwnProperty('Ref')) {
        obj['Ref'] = PRIndirectReference.constructFromObject(data['Ref']);
      }
    }
    return obj;
  }

  /**
   * @member {String} WidgetName
   */
  exports.prototype['WidgetName'] = undefined;
  /**
   * @member {String} Name
   */
  exports.prototype['Name'] = undefined;
  /**
   * @member {module:model/PdfDictionary} Info
   */
  exports.prototype['Info'] = undefined;
  /**
   * @member {module:model/PRIndirectReference} Ref
   */
  exports.prototype['Ref'] = undefined;



  return exports;
}));
