// mslpsapi/src/model/Order.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/PdfPTable'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./PdfPTable'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.Order = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.PdfPTable);
  }
}(this, function(ApiClient, PdfPTable) {
  'use strict';




  /**
   * The Order model module.
   * @module model/Order
   * @version v1
   */

  /**
   * Constructs a new <code>Order</code>.
   * @alias module:model/Order
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>Order</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Order} obj Optional instance to populate.
   * @return {module:model/Order} The populated <code>Order</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('LogisticsNo')) {
        obj['LogisticsNo'] = ApiClient.convertToType(data['LogisticsNo'], 'String');
      }
      if (data.hasOwnProperty('PickingInfo')) {
        obj['PickingInfo'] = ApiClient.convertToType(data['PickingInfo'], 'String');
      }
      if (data.hasOwnProperty('PickingTemplate')) {
        obj['PickingTemplate'] = PdfPTable.constructFromObject(data['PickingTemplate']);
      }
    }
    return obj;
  }

  /**
   * @member {String} LogisticsNo
   */
  exports.prototype['LogisticsNo'] = undefined;
  /**
   * @member {String} PickingInfo
   */
  exports.prototype['PickingInfo'] = undefined;
  /**
   * @member {module:model/PdfPTable} PickingTemplate
   */
  exports.prototype['PickingTemplate'] = undefined;



  return exports;
}));
