// mslpsapi/src/model/OrderInfo.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/Address', 'model/ProductInfo'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./Address'), require('./ProductInfo'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.OrderInfo = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.Address, root.Mslpsapi.ProductInfo);
  }
}(this, function(ApiClient, Address, ProductInfo) {
  'use strict';




  /**
   * The OrderInfo model module.
   * @module model/OrderInfo
   * @version v1
   */

  /**
   * Constructs a new <code>OrderInfo</code>.
   * @alias module:model/OrderInfo
   * @class
   */
  var exports = function() {
    var _this = this;




























  };

  /**
   * Constructs a <code>OrderInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/OrderInfo} obj Optional instance to populate.
   * @return {module:model/OrderInfo} The populated <code>OrderInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('ProductInfos')) {
        obj['ProductInfos'] = ApiClient.convertToType(data['ProductInfos'], [ProductInfo]);
      }
      if (data.hasOwnProperty('OrderNumber')) {
        obj['OrderNumber'] = ApiClient.convertToType(data['OrderNumber'], 'String');
      }
      if (data.hasOwnProperty('LogisticsNo')) {
        obj['LogisticsNo'] = ApiClient.convertToType(data['LogisticsNo'], 'String');
      }
      if (data.hasOwnProperty('Comment')) {
        obj['Comment'] = ApiClient.convertToType(data['Comment'], 'String');
      }
      if (data.hasOwnProperty('Weight')) {
        obj['Weight'] = ApiClient.convertToType(data['Weight'], 'Number');
      }
      if (data.hasOwnProperty('Price')) {
        obj['Price'] = ApiClient.convertToType(data['Price'], 'Number');
      }
      if (data.hasOwnProperty('Contact')) {
        obj['Contact'] = ApiClient.convertToType(data['Contact'], 'String');
      }
      if (data.hasOwnProperty('Company')) {
        obj['Company'] = ApiClient.convertToType(data['Company'], 'String');
      }
      if (data.hasOwnProperty('Tel')) {
        obj['Tel'] = ApiClient.convertToType(data['Tel'], 'String');
      }
      if (data.hasOwnProperty('MobilePhone')) {
        obj['MobilePhone'] = ApiClient.convertToType(data['MobilePhone'], 'String');
      }
      if (data.hasOwnProperty('CountryCode')) {
        obj['CountryCode'] = ApiClient.convertToType(data['CountryCode'], 'String');
      }
      if (data.hasOwnProperty('CountryCn')) {
        obj['CountryCn'] = ApiClient.convertToType(data['CountryCn'], 'String');
      }
      if (data.hasOwnProperty('CountryEn')) {
        obj['CountryEn'] = ApiClient.convertToType(data['CountryEn'], 'String');
      }
      if (data.hasOwnProperty('Province')) {
        obj['Province'] = ApiClient.convertToType(data['Province'], 'String');
      }
      if (data.hasOwnProperty('City')) {
        obj['City'] = ApiClient.convertToType(data['City'], 'String');
      }
      if (data.hasOwnProperty('Country')) {
        obj['Country'] = ApiClient.convertToType(data['Country'], 'String');
      }
      if (data.hasOwnProperty('Address1')) {
        obj['Address1'] = ApiClient.convertToType(data['Address1'], 'String');
      }
      if (data.hasOwnProperty('Address2')) {
        obj['Address2'] = ApiClient.convertToType(data['Address2'], 'String');
      }
      if (data.hasOwnProperty('ZipCode')) {
        obj['ZipCode'] = ApiClient.convertToType(data['ZipCode'], 'String');
      }
      if (data.hasOwnProperty('Email')) {
        obj['Email'] = ApiClient.convertToType(data['Email'], 'String');
      }
      if (data.hasOwnProperty('Fax')) {
        obj['Fax'] = ApiClient.convertToType(data['Fax'], 'String');
      }
      if (data.hasOwnProperty('Quantity')) {
        obj['Quantity'] = ApiClient.convertToType(data['Quantity'], 'Integer');
      }
      if (data.hasOwnProperty('EShopIdentity')) {
        obj['EShopIdentity'] = ApiClient.convertToType(data['EShopIdentity'], 'String');
      }
      if (data.hasOwnProperty('OriginalNo')) {
        obj['OriginalNo'] = ApiClient.convertToType(data['OriginalNo'], 'String');
      }
      if (data.hasOwnProperty('SendAddress')) {
        obj['SendAddress'] = Address.constructFromObject(data['SendAddress']);
      }
      if (data.hasOwnProperty('ReturnAddress')) {
        obj['ReturnAddress'] = Address.constructFromObject(data['ReturnAddress']);
      }
      if (data.hasOwnProperty('PickupAddress')) {
        obj['PickupAddress'] = Address.constructFromObject(data['PickupAddress']);
      }
    }
    return obj;
  }

  /**
   * @member {Array.<module:model/ProductInfo>} ProductInfos
   */
  exports.prototype['ProductInfos'] = undefined;
  /**
   * @member {String} OrderNumber
   */
  exports.prototype['OrderNumber'] = undefined;
  /**
   * @member {String} LogisticsNo
   */
  exports.prototype['LogisticsNo'] = undefined;
  /**
   * @member {String} Comment
   */
  exports.prototype['Comment'] = undefined;
  /**
   * @member {Number} Weight
   */
  exports.prototype['Weight'] = undefined;
  /**
   * @member {Number} Price
   */
  exports.prototype['Price'] = undefined;
  /**
   * @member {String} Contact
   */
  exports.prototype['Contact'] = undefined;
  /**
   * @member {String} Company
   */
  exports.prototype['Company'] = undefined;
  /**
   * @member {String} Tel
   */
  exports.prototype['Tel'] = undefined;
  /**
   * @member {String} MobilePhone
   */
  exports.prototype['MobilePhone'] = undefined;
  /**
   * @member {String} CountryCode
   */
  exports.prototype['CountryCode'] = undefined;
  /**
   * @member {String} CountryCn
   */
  exports.prototype['CountryCn'] = undefined;
  /**
   * @member {String} CountryEn
   */
  exports.prototype['CountryEn'] = undefined;
  /**
   * @member {String} Province
   */
  exports.prototype['Province'] = undefined;
  /**
   * @member {String} City
   */
  exports.prototype['City'] = undefined;
  /**
   * @member {String} Country
   */
  exports.prototype['Country'] = undefined;
  /**
   * @member {String} Address1
   */
  exports.prototype['Address1'] = undefined;
  /**
   * @member {String} Address2
   */
  exports.prototype['Address2'] = undefined;
  /**
   * @member {String} ZipCode
   */
  exports.prototype['ZipCode'] = undefined;
  /**
   * @member {String} Email
   */
  exports.prototype['Email'] = undefined;
  /**
   * @member {String} Fax
   */
  exports.prototype['Fax'] = undefined;
  /**
   * @member {Integer} Quantity
   */
  exports.prototype['Quantity'] = undefined;
  /**
   * @member {String} EShopIdentity
   */
  exports.prototype['EShopIdentity'] = undefined;
  /**
   * @member {String} OriginalNo
   */
  exports.prototype['OriginalNo'] = undefined;
  /**
   * @member {module:model/Address} SendAddress
   */
  exports.prototype['SendAddress'] = undefined;
  /**
   * @member {module:model/Address} ReturnAddress
   */
  exports.prototype['ReturnAddress'] = undefined;
  /**
   * @member {module:model/Address} PickupAddress
   */
  exports.prototype['PickupAddress'] = undefined;



  return exports;
}));
