// mslpsapi/src/model/PdfCollection.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/PRIndirectReference', 'model/PdfCollectionSchema', 'model/PdfCollectionSort', 'model/PdfName'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./PRIndirectReference'), require('./PdfCollectionSchema'), require('./PdfCollectionSort'), require('./PdfName'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.PdfCollection = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.PRIndirectReference, root.Mslpsapi.PdfCollectionSchema, root.Mslpsapi.PdfCollectionSort, root.Mslpsapi.PdfName);
  }
}(this, function(ApiClient, PRIndirectReference, PdfCollectionSchema, PdfCollectionSort, PdfName) {
  'use strict';




  /**
   * The PdfCollection model module.
   * @module model/PdfCollection
   * @version v1
   */

  /**
   * Constructs a new <code>PdfCollection</code>.
   * @alias module:model/PdfCollection
   * @class
   */
  var exports = function() {
    var _this = this;









  };

  /**
   * Constructs a <code>PdfCollection</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PdfCollection} obj Optional instance to populate.
   * @return {module:model/PdfCollection} The populated <code>PdfCollection</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('InitialDocument')) {
        obj['InitialDocument'] = ApiClient.convertToType(data['InitialDocument'], 'String');
      }
      if (data.hasOwnProperty('Schema')) {
        obj['Schema'] = PdfCollectionSchema.constructFromObject(data['Schema']);
      }
      if (data.hasOwnProperty('Sort')) {
        obj['Sort'] = PdfCollectionSort.constructFromObject(data['Sort']);
      }
      if (data.hasOwnProperty('Keys')) {
        obj['Keys'] = ApiClient.convertToType(data['Keys'], [PdfName]);
      }
      if (data.hasOwnProperty('Size')) {
        obj['Size'] = ApiClient.convertToType(data['Size'], 'Integer');
      }
      if (data.hasOwnProperty('Length')) {
        obj['Length'] = ApiClient.convertToType(data['Length'], 'Integer');
      }
      if (data.hasOwnProperty('Type')) {
        obj['Type'] = ApiClient.convertToType(data['Type'], 'Integer');
      }
      if (data.hasOwnProperty('IndRef')) {
        obj['IndRef'] = PRIndirectReference.constructFromObject(data['IndRef']);
      }
    }
    return obj;
  }

  /**
   * @member {String} InitialDocument
   */
  exports.prototype['InitialDocument'] = undefined;
  /**
   * @member {module:model/PdfCollectionSchema} Schema
   */
  exports.prototype['Schema'] = undefined;
  /**
   * @member {module:model/PdfCollectionSort} Sort
   */
  exports.prototype['Sort'] = undefined;
  /**
   * @member {Array.<module:model/PdfName>} Keys
   */
  exports.prototype['Keys'] = undefined;
  /**
   * @member {Integer} Size
   */
  exports.prototype['Size'] = undefined;
  /**
   * @member {Integer} Length
   */
  exports.prototype['Length'] = undefined;
  /**
   * @member {Integer} Type
   */
  exports.prototype['Type'] = undefined;
  /**
   * @member {module:model/PRIndirectReference} IndRef
   */
  exports.prototype['IndRef'] = undefined;



  return exports;
}));
