// mslpsapi/src/model/PdfContentByte.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/ByteBuffer', 'model/PdfContentByte', 'model/PdfDocument', 'model/PdfOutline', 'model/PdfWriter'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./ByteBuffer'), require('./PdfContentByte'), require('./PdfDocument'), require('./PdfOutline'), require('./PdfWriter'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.PdfContentByte = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.ByteBuffer, root.Mslpsapi.PdfContentByte, root.Mslpsapi.PdfDocument, root.Mslpsapi.PdfOutline, root.Mslpsapi.PdfWriter);
  }
}(this, function(ApiClient, ByteBuffer, PdfContentByte, PdfDocument, PdfOutline, PdfWriter) {
  'use strict';




  /**
   * The PdfContentByte model module.
   * @module model/PdfContentByte
   * @version v1
   */

  /**
   * Constructs a new <code>PdfContentByte</code>.
   * @alias module:model/PdfContentByte
   * @class
   */
  var exports = function() {
    var _this = this;













  };

  /**
   * Constructs a <code>PdfContentByte</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PdfContentByte} obj Optional instance to populate.
   * @return {module:model/PdfContentByte} The populated <code>PdfContentByte</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('SuppressTagging')) {
        obj['SuppressTagging'] = ApiClient.convertToType(data['SuppressTagging'], 'Boolean');
      }
      if (data.hasOwnProperty('InternalBuffer')) {
        obj['InternalBuffer'] = ByteBuffer.constructFromObject(data['InternalBuffer']);
      }
      if (data.hasOwnProperty('XTLM')) {
        obj['XTLM'] = ApiClient.convertToType(data['XTLM'], 'Number');
      }
      if (data.hasOwnProperty('YTLM')) {
        obj['YTLM'] = ApiClient.convertToType(data['YTLM'], 'Number');
      }
      if (data.hasOwnProperty('CharacterSpacing')) {
        obj['CharacterSpacing'] = ApiClient.convertToType(data['CharacterSpacing'], 'Number');
      }
      if (data.hasOwnProperty('WordSpacing')) {
        obj['WordSpacing'] = ApiClient.convertToType(data['WordSpacing'], 'Number');
      }
      if (data.hasOwnProperty('HorizontalScaling')) {
        obj['HorizontalScaling'] = ApiClient.convertToType(data['HorizontalScaling'], 'Number');
      }
      if (data.hasOwnProperty('Leading')) {
        obj['Leading'] = ApiClient.convertToType(data['Leading'], 'Number');
      }
      if (data.hasOwnProperty('RootOutline')) {
        obj['RootOutline'] = PdfOutline.constructFromObject(data['RootOutline']);
      }
      if (data.hasOwnProperty('PdfWriter')) {
        obj['PdfWriter'] = PdfWriter.constructFromObject(data['PdfWriter']);
      }
      if (data.hasOwnProperty('PdfDocument')) {
        obj['PdfDocument'] = PdfDocument.constructFromObject(data['PdfDocument']);
      }
      if (data.hasOwnProperty('Duplicate')) {
        obj['Duplicate'] = PdfContentByte.constructFromObject(data['Duplicate']);
      }
    }
    return obj;
  }

  /**
   * @member {Boolean} SuppressTagging
   */
  exports.prototype['SuppressTagging'] = undefined;
  /**
   * @member {module:model/ByteBuffer} InternalBuffer
   */
  exports.prototype['InternalBuffer'] = undefined;
  /**
   * @member {Number} XTLM
   */
  exports.prototype['XTLM'] = undefined;
  /**
   * @member {Number} YTLM
   */
  exports.prototype['YTLM'] = undefined;
  /**
   * @member {Number} CharacterSpacing
   */
  exports.prototype['CharacterSpacing'] = undefined;
  /**
   * @member {Number} WordSpacing
   */
  exports.prototype['WordSpacing'] = undefined;
  /**
   * @member {Number} HorizontalScaling
   */
  exports.prototype['HorizontalScaling'] = undefined;
  /**
   * @member {Number} Leading
   */
  exports.prototype['Leading'] = undefined;
  /**
   * @member {module:model/PdfOutline} RootOutline
   */
  exports.prototype['RootOutline'] = undefined;
  /**
   * @member {module:model/PdfWriter} PdfWriter
   */
  exports.prototype['PdfWriter'] = undefined;
  /**
   * @member {module:model/PdfDocument} PdfDocument
   */
  exports.prototype['PdfDocument'] = undefined;
  /**
   * @member {module:model/PdfContentByte} Duplicate
   */
  exports.prototype['Duplicate'] = undefined;



  return exports;
}));
