// mslpsapi/src/model/PdfOutline.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/BaseColor', 'model/PRIndirectReference', 'model/PdfIndirectReference', 'model/PdfName', 'model/PdfObject', 'model/PdfOutline'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./BaseColor'), require('./PRIndirectReference'), require('./PdfIndirectReference'), require('./PdfName'), require('./PdfObject'), require('./PdfOutline'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.PdfOutline = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.BaseColor, root.Mslpsapi.PRIndirectReference, root.Mslpsapi.PdfIndirectReference, root.Mslpsapi.PdfName, root.Mslpsapi.PdfObject, root.Mslpsapi.PdfOutline);
  }
}(this, function(ApiClient, BaseColor, PRIndirectReference, PdfIndirectReference, PdfName, PdfObject, PdfOutline) {
  'use strict';




  /**
   * The PdfOutline model module.
   * @module model/PdfOutline
   * @version v1
   */

  /**
   * Constructs a new <code>PdfOutline</code>.
   * @alias module:model/PdfOutline
   * @class
   */
  var exports = function() {
    var _this = this;
















  };

  /**
   * Constructs a <code>PdfOutline</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PdfOutline} obj Optional instance to populate.
   * @return {module:model/PdfOutline} The populated <code>PdfOutline</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('IndirectReference')) {
        obj['IndirectReference'] = PdfIndirectReference.constructFromObject(data['IndirectReference']);
      }
      if (data.hasOwnProperty('Parent')) {
        obj['Parent'] = PdfOutline.constructFromObject(data['Parent']);
      }
      if (data.hasOwnProperty('PdfDestination')) {
        obj['PdfDestination'] = ApiClient.convertToType(data['PdfDestination'], [PdfObject]);
      }
      if (data.hasOwnProperty('Level')) {
        obj['Level'] = ApiClient.convertToType(data['Level'], 'Integer');
      }
      if (data.hasOwnProperty('Kids')) {
        obj['Kids'] = ApiClient.convertToType(data['Kids'], [PdfOutline]);
      }
      if (data.hasOwnProperty('Tag')) {
        obj['Tag'] = ApiClient.convertToType(data['Tag'], 'String');
      }
      if (data.hasOwnProperty('Title')) {
        obj['Title'] = ApiClient.convertToType(data['Title'], 'String');
      }
      if (data.hasOwnProperty('Open')) {
        obj['Open'] = ApiClient.convertToType(data['Open'], 'Boolean');
      }
      if (data.hasOwnProperty('Color')) {
        obj['Color'] = BaseColor.constructFromObject(data['Color']);
      }
      if (data.hasOwnProperty('Style')) {
        obj['Style'] = ApiClient.convertToType(data['Style'], 'Integer');
      }
      if (data.hasOwnProperty('Keys')) {
        obj['Keys'] = ApiClient.convertToType(data['Keys'], [PdfName]);
      }
      if (data.hasOwnProperty('Size')) {
        obj['Size'] = ApiClient.convertToType(data['Size'], 'Integer');
      }
      if (data.hasOwnProperty('Length')) {
        obj['Length'] = ApiClient.convertToType(data['Length'], 'Integer');
      }
      if (data.hasOwnProperty('Type')) {
        obj['Type'] = ApiClient.convertToType(data['Type'], 'Integer');
      }
      if (data.hasOwnProperty('IndRef')) {
        obj['IndRef'] = PRIndirectReference.constructFromObject(data['IndRef']);
      }
    }
    return obj;
  }

  /**
   * @member {module:model/PdfIndirectReference} IndirectReference
   */
  exports.prototype['IndirectReference'] = undefined;
  /**
   * @member {module:model/PdfOutline} Parent
   */
  exports.prototype['Parent'] = undefined;
  /**
   * @member {Array.<module:model/PdfObject>} PdfDestination
   */
  exports.prototype['PdfDestination'] = undefined;
  /**
   * @member {Integer} Level
   */
  exports.prototype['Level'] = undefined;
  /**
   * @member {Array.<module:model/PdfOutline>} Kids
   */
  exports.prototype['Kids'] = undefined;
  /**
   * @member {String} Tag
   */
  exports.prototype['Tag'] = undefined;
  /**
   * @member {String} Title
   */
  exports.prototype['Title'] = undefined;
  /**
   * @member {Boolean} Open
   */
  exports.prototype['Open'] = undefined;
  /**
   * @member {module:model/BaseColor} Color
   */
  exports.prototype['Color'] = undefined;
  /**
   * @member {Integer} Style
   */
  exports.prototype['Style'] = undefined;
  /**
   * @member {Array.<module:model/PdfName>} Keys
   */
  exports.prototype['Keys'] = undefined;
  /**
   * @member {Integer} Size
   */
  exports.prototype['Size'] = undefined;
  /**
   * @member {Integer} Length
   */
  exports.prototype['Length'] = undefined;
  /**
   * @member {Integer} Type
   */
  exports.prototype['Type'] = undefined;
  /**
   * @member {module:model/PRIndirectReference} IndRef
   */
  exports.prototype['IndRef'] = undefined;



  return exports;
}));
