// mslpsapi/src/model/PdfPRow.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/AccessibleElementId', 'model/PdfName'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./AccessibleElementId'), require('./PdfName'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.PdfPRow = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.AccessibleElementId, root.Mslpsapi.PdfName);
  }
}(this, function(ApiClient, AccessibleElementId, PdfName) {
  'use strict';




  /**
   * The PdfPRow model module.
   * @module model/PdfPRow
   * @version v1
   */

  /**
   * Constructs a new <code>PdfPRow</code>.
   * @alias module:model/PdfPRow
   * @class
   */
  var exports = function() {
    var _this = this;








  };

  /**
   * Constructs a <code>PdfPRow</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PdfPRow} obj Optional instance to populate.
   * @return {module:model/PdfPRow} The populated <code>PdfPRow</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('mayNotBreak')) {
        obj['mayNotBreak'] = ApiClient.convertToType(data['mayNotBreak'], 'Boolean');
      }
      if (data.hasOwnProperty('MayNotBreak')) {
        obj['MayNotBreak'] = ApiClient.convertToType(data['MayNotBreak'], 'Boolean');
      }
      if (data.hasOwnProperty('MaxHeights')) {
        obj['MaxHeights'] = ApiClient.convertToType(data['MaxHeights'], 'Number');
      }
      if (data.hasOwnProperty('Adjusted')) {
        obj['Adjusted'] = ApiClient.convertToType(data['Adjusted'], 'Boolean');
      }
      if (data.hasOwnProperty('Role')) {
        obj['Role'] = PdfName.constructFromObject(data['Role']);
      }
      if (data.hasOwnProperty('ID')) {
        obj['ID'] = AccessibleElementId.constructFromObject(data['ID']);
      }
      if (data.hasOwnProperty('IsInline')) {
        obj['IsInline'] = ApiClient.convertToType(data['IsInline'], 'Boolean');
      }
    }
    return obj;
  }

  /**
   * @member {Boolean} mayNotBreak
   */
  exports.prototype['mayNotBreak'] = undefined;
  /**
   * @member {Boolean} MayNotBreak
   */
  exports.prototype['MayNotBreak'] = undefined;
  /**
   * @member {Number} MaxHeights
   */
  exports.prototype['MaxHeights'] = undefined;
  /**
   * @member {Boolean} Adjusted
   */
  exports.prototype['Adjusted'] = undefined;
  /**
   * @member {module:model/PdfName} Role
   */
  exports.prototype['Role'] = undefined;
  /**
   * @member {module:model/AccessibleElementId} ID
   */
  exports.prototype['ID'] = undefined;
  /**
   * @member {Boolean} IsInline
   */
  exports.prototype['IsInline'] = undefined;



  return exports;
}));
