// mslpsapi/src/model/PdfTransition.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/PdfDictionary'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./PdfDictionary'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.PdfTransition = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.PdfDictionary);
  }
}(this, function(ApiClient, PdfDictionary) {
  'use strict';




  /**
   * The PdfTransition model module.
   * @module model/PdfTransition
   * @version v1
   */

  /**
   * Constructs a new <code>PdfTransition</code>.
   * @alias module:model/PdfTransition
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>PdfTransition</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PdfTransition} obj Optional instance to populate.
   * @return {module:model/PdfTransition} The populated <code>PdfTransition</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('Duration')) {
        obj['Duration'] = ApiClient.convertToType(data['Duration'], 'Integer');
      }
      if (data.hasOwnProperty('Type')) {
        obj['Type'] = ApiClient.convertToType(data['Type'], 'Integer');
      }
      if (data.hasOwnProperty('TransitionDictionary')) {
        obj['TransitionDictionary'] = PdfDictionary.constructFromObject(data['TransitionDictionary']);
      }
    }
    return obj;
  }

  /**
   * @member {Integer} Duration
   */
  exports.prototype['Duration'] = undefined;
  /**
   * @member {Integer} Type
   */
  exports.prototype['Type'] = undefined;
  /**
   * @member {module:model/PdfDictionary} TransitionDictionary
   */
  exports.prototype['TransitionDictionary'] = undefined;



  return exports;
}));
