// mslpsapi/src/model/PrintInfo.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/ConfigInfo', 'model/Order', 'model/PdfPTable'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./ConfigInfo'), require('./Order'), require('./PdfPTable'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.PrintInfo = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.ConfigInfo, root.Mslpsapi.Order, root.Mslpsapi.PdfPTable);
  }
}(this, function(ApiClient, ConfigInfo, Order, PdfPTable) {
  'use strict';




  /**
   * The PrintInfo model module.
   * @module model/PrintInfo
   * @version v1
   */

  /**
   * Constructs a new <code>PrintInfo</code>.
   * @alias module:model/PrintInfo
   * @class
   */
  var exports = function() {
    var _this = this;









  };

  /**
   * Constructs a <code>PrintInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PrintInfo} obj Optional instance to populate.
   * @return {module:model/PrintInfo} The populated <code>PrintInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('Identity')) {
        obj['Identity'] = ApiClient.convertToType(data['Identity'], 'String');
      }
      if (data.hasOwnProperty('Channel')) {
        obj['Channel'] = ApiClient.convertToType(data['Channel'], 'String');
      }
      if (data.hasOwnProperty('Country')) {
        obj['Country'] = ApiClient.convertToType(data['Country'], 'String');
      }
      if (data.hasOwnProperty('EShop')) {
        obj['EShop'] = ApiClient.convertToType(data['EShop'], 'String');
      }
      if (data.hasOwnProperty('SourceTemplate')) {
        obj['SourceTemplate'] = PdfPTable.constructFromObject(data['SourceTemplate']);
      }
      if (data.hasOwnProperty('Configs')) {
        obj['Configs'] = ApiClient.convertToType(data['Configs'], [ConfigInfo]);
      }
      if (data.hasOwnProperty('Orders')) {
        obj['Orders'] = ApiClient.convertToType(data['Orders'], [Order]);
      }
      if (data.hasOwnProperty('PickingInfo')) {
        obj['PickingInfo'] = ApiClient.convertToType(data['PickingInfo'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} Identity
   */
  exports.prototype['Identity'] = undefined;
  /**
   * @member {String} Channel
   */
  exports.prototype['Channel'] = undefined;
  /**
   * @member {String} Country
   */
  exports.prototype['Country'] = undefined;
  /**
   * @member {String} EShop
   */
  exports.prototype['EShop'] = undefined;
  /**
   * @member {module:model/PdfPTable} SourceTemplate
   */
  exports.prototype['SourceTemplate'] = undefined;
  /**
   * @member {Array.<module:model/ConfigInfo>} Configs
   */
  exports.prototype['Configs'] = undefined;
  /**
   * @member {Array.<module:model/Order>} Orders
   */
  exports.prototype['Orders'] = undefined;
  /**
   * @member {String} PickingInfo
   */
  exports.prototype['PickingInfo'] = undefined;



  return exports;
}));
