// mslpsapi/src/model/ProductInfo.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.ProductInfo = factory(root.Mslpsapi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The ProductInfo model module.
   * @module model/ProductInfo
   * @version v1
   */

  /**
   * Constructs a new <code>ProductInfo</code>.
   * @alias module:model/ProductInfo
   * @class
   */
  var exports = function() {
    var _this = this;























  };

  /**
   * Constructs a <code>ProductInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ProductInfo} obj Optional instance to populate.
   * @return {module:model/ProductInfo} The populated <code>ProductInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('ProductId')) {
        obj['ProductId'] = ApiClient.convertToType(data['ProductId'], 'Integer');
      }
      if (data.hasOwnProperty('EnTradeName')) {
        obj['EnTradeName'] = ApiClient.convertToType(data['EnTradeName'], 'String');
      }
      if (data.hasOwnProperty('CnTradeName')) {
        obj['CnTradeName'] = ApiClient.convertToType(data['CnTradeName'], 'String');
      }
      if (data.hasOwnProperty('Properties')) {
        obj['Properties'] = ApiClient.convertToType(data['Properties'], 'String');
      }
      if (data.hasOwnProperty('OwerOrder')) {
        obj['OwerOrder'] = ApiClient.convertToType(data['OwerOrder'], 'String');
      }
      if (data.hasOwnProperty('Title')) {
        obj['Title'] = ApiClient.convertToType(data['Title'], 'String');
      }
      if (data.hasOwnProperty('SkuCode')) {
        obj['SkuCode'] = ApiClient.convertToType(data['SkuCode'], 'String');
      }
      if (data.hasOwnProperty('HsCode')) {
        obj['HsCode'] = ApiClient.convertToType(data['HsCode'], 'String');
      }
      if (data.hasOwnProperty('Quantity')) {
        obj['Quantity'] = ApiClient.convertToType(data['Quantity'], 'Integer');
      }
      if (data.hasOwnProperty('Weight')) {
        obj['Weight'] = ApiClient.convertToType(data['Weight'], 'Number');
      }
      if (data.hasOwnProperty('DeclareValue')) {
        obj['DeclareValue'] = ApiClient.convertToType(data['DeclareValue'], 'Number');
      }
      if (data.hasOwnProperty('Comment')) {
        obj['Comment'] = ApiClient.convertToType(data['Comment'], 'String');
      }
      if (data.hasOwnProperty('IsFragile')) {
        obj['IsFragile'] = ApiClient.convertToType(data['IsFragile'], 'Boolean');
      }
      if (data.hasOwnProperty('IsLiquid')) {
        obj['IsLiquid'] = ApiClient.convertToType(data['IsLiquid'], 'Boolean');
      }
      if (data.hasOwnProperty('IsPureBattery')) {
        obj['IsPureBattery'] = ApiClient.convertToType(data['IsPureBattery'], 'Boolean');
      }
      if (data.hasOwnProperty('IsLithiumBattery')) {
        obj['IsLithiumBattery'] = ApiClient.convertToType(data['IsLithiumBattery'], 'Boolean');
      }
      if (data.hasOwnProperty('TaxNumber')) {
        obj['TaxNumber'] = ApiClient.convertToType(data['TaxNumber'], 'String');
      }
      if (data.hasOwnProperty('ImageUrl')) {
        obj['ImageUrl'] = ApiClient.convertToType(data['ImageUrl'], 'String');
      }
      if (data.hasOwnProperty('LinkUrl')) {
        obj['LinkUrl'] = ApiClient.convertToType(data['LinkUrl'], 'String');
      }
      if (data.hasOwnProperty('Origin')) {
        obj['Origin'] = ApiClient.convertToType(data['Origin'], 'String');
      }
      if (data.hasOwnProperty('Currency')) {
        obj['Currency'] = ApiClient.convertToType(data['Currency'], 'String');
      }
      if (data.hasOwnProperty('Uint')) {
        obj['Uint'] = ApiClient.convertToType(data['Uint'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {Integer} ProductId
   */
  exports.prototype['ProductId'] = undefined;
  /**
   * @member {String} EnTradeName
   */
  exports.prototype['EnTradeName'] = undefined;
  /**
   * @member {String} CnTradeName
   */
  exports.prototype['CnTradeName'] = undefined;
  /**
   * @member {String} Properties
   */
  exports.prototype['Properties'] = undefined;
  /**
   * @member {String} OwerOrder
   */
  exports.prototype['OwerOrder'] = undefined;
  /**
   * @member {String} Title
   */
  exports.prototype['Title'] = undefined;
  /**
   * @member {String} SkuCode
   */
  exports.prototype['SkuCode'] = undefined;
  /**
   * @member {String} HsCode
   */
  exports.prototype['HsCode'] = undefined;
  /**
   * @member {Integer} Quantity
   */
  exports.prototype['Quantity'] = undefined;
  /**
   * @member {Number} Weight
   */
  exports.prototype['Weight'] = undefined;
  /**
   * @member {Number} DeclareValue
   */
  exports.prototype['DeclareValue'] = undefined;
  /**
   * @member {String} Comment
   */
  exports.prototype['Comment'] = undefined;
  /**
   * @member {Boolean} IsFragile
   */
  exports.prototype['IsFragile'] = undefined;
  /**
   * @member {Boolean} IsLiquid
   */
  exports.prototype['IsLiquid'] = undefined;
  /**
   * @member {Boolean} IsPureBattery
   */
  exports.prototype['IsPureBattery'] = undefined;
  /**
   * @member {Boolean} IsLithiumBattery
   */
  exports.prototype['IsLithiumBattery'] = undefined;
  /**
   * @member {String} TaxNumber
   */
  exports.prototype['TaxNumber'] = undefined;
  /**
   * @member {String} ImageUrl
   */
  exports.prototype['ImageUrl'] = undefined;
  /**
   * @member {String} LinkUrl
   */
  exports.prototype['LinkUrl'] = undefined;
  /**
   * @member {String} Origin
   */
  exports.prototype['Origin'] = undefined;
  /**
   * @member {String} Currency
   */
  exports.prototype['Currency'] = undefined;
  /**
   * @member {String} Uint
   */
  exports.prototype['Uint'] = undefined;



  return exports;
}));
