// mslpsapi/src/model/TabStop.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/IDrawInterface'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./IDrawInterface'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.TabStop = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.IDrawInterface);
  }
}(this, function(ApiClient, IDrawInterface) {
  'use strict';




  /**
   * The TabStop model module.
   * @module model/TabStop
   * @version v1
   */

  /**
   * Constructs a new <code>TabStop</code>.
   * @alias module:model/TabStop
   * @class
   */
  var exports = function() {
    var _this = this;





  };

  /**
   * Constructs a <code>TabStop</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/TabStop} obj Optional instance to populate.
   * @return {module:model/TabStop} The populated <code>TabStop</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('Position')) {
        obj['Position'] = ApiClient.convertToType(data['Position'], 'Number');
      }
      if (data.hasOwnProperty('Align')) {
        obj['Align'] = ApiClient.convertToType(data['Align'], 'Integer');
      }
      if (data.hasOwnProperty('Leader')) {
        obj['Leader'] = IDrawInterface.constructFromObject(data['Leader']);
      }
      if (data.hasOwnProperty('AnchorChar')) {
        obj['AnchorChar'] = ApiClient.convertToType(data['AnchorChar'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {Number} Position
   */
  exports.prototype['Position'] = undefined;
  /**
   * @member {module:model/TabStop.AlignEnum} Align
   */
  exports.prototype['Align'] = undefined;
  /**
   * @member {module:model/IDrawInterface} Leader
   */
  exports.prototype['Leader'] = undefined;
  /**
   * @member {String} AnchorChar
   */
  exports.prototype['AnchorChar'] = undefined;


  /**
   * Allowed values for the <code>Align</code> property.
   * @enum {Integer}
   * @readonly
   */
  exports.AlignEnum = {
    /**
     * value: 0
     * @const
     */
    "0": 0,
    /**
     * value: 1
     * @const
     */
    "1": 1,
    /**
     * value: 2
     * @const
     */
    "2": 2,
    /**
     * value: 3
     * @const
     */
    "3": 3  };


  return exports;
}));
