// mslpsapi/src/model/XfaForm.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/AcroFieldsSearch', 'model/PdfReader', 'model/Xml2SomDatasets', 'model/Xml2SomTemplate'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./AcroFieldsSearch'), require('./PdfReader'), require('./Xml2SomDatasets'), require('./Xml2SomTemplate'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.XfaForm = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.AcroFieldsSearch, root.Mslpsapi.PdfReader, root.Mslpsapi.Xml2SomDatasets, root.Mslpsapi.Xml2SomTemplate);
  }
}(this, function(ApiClient, AcroFieldsSearch, PdfReader, Xml2SomDatasets, Xml2SomTemplate) {
  'use strict';




  /**
   * The XfaForm model module.
   * @module model/XfaForm
   * @version v1
   */

  /**
   * Constructs a new <code>XfaForm</code>.
   * @alias module:model/XfaForm
   * @class
   */
  var exports = function() {
    var _this = this;









  };

  /**
   * Constructs a <code>XfaForm</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/XfaForm} obj Optional instance to populate.
   * @return {module:model/XfaForm} The populated <code>XfaForm</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('XfaPresent')) {
        obj['XfaPresent'] = ApiClient.convertToType(data['XfaPresent'], 'Boolean');
      }
      if (data.hasOwnProperty('DomDocument')) {
        obj['DomDocument'] = ApiClient.convertToType(data['DomDocument'], [Object]);
      }
      if (data.hasOwnProperty('Reader')) {
        obj['Reader'] = PdfReader.constructFromObject(data['Reader']);
      }
      if (data.hasOwnProperty('Changed')) {
        obj['Changed'] = ApiClient.convertToType(data['Changed'], 'Boolean');
      }
      if (data.hasOwnProperty('TemplateSom')) {
        obj['TemplateSom'] = Xml2SomTemplate.constructFromObject(data['TemplateSom']);
      }
      if (data.hasOwnProperty('DatasetsSom')) {
        obj['DatasetsSom'] = Xml2SomDatasets.constructFromObject(data['DatasetsSom']);
      }
      if (data.hasOwnProperty('AcroFieldsSom')) {
        obj['AcroFieldsSom'] = AcroFieldsSearch.constructFromObject(data['AcroFieldsSom']);
      }
      if (data.hasOwnProperty('DatasetsNode')) {
        obj['DatasetsNode'] = ApiClient.convertToType(data['DatasetsNode'], [Object]);
      }
    }
    return obj;
  }

  /**
   * @member {Boolean} XfaPresent
   */
  exports.prototype['XfaPresent'] = undefined;
  /**
   * @member {Array.<Object>} DomDocument
   */
  exports.prototype['DomDocument'] = undefined;
  /**
   * @member {module:model/PdfReader} Reader
   */
  exports.prototype['Reader'] = undefined;
  /**
   * @member {Boolean} Changed
   */
  exports.prototype['Changed'] = undefined;
  /**
   * @member {module:model/Xml2SomTemplate} TemplateSom
   */
  exports.prototype['TemplateSom'] = undefined;
  /**
   * @member {module:model/Xml2SomDatasets} DatasetsSom
   */
  exports.prototype['DatasetsSom'] = undefined;
  /**
   * @member {module:model/AcroFieldsSearch} AcroFieldsSom
   */
  exports.prototype['AcroFieldsSom'] = undefined;
  /**
   * @member {Array.<Object>} DatasetsNode
   */
  exports.prototype['DatasetsNode'] = undefined;



  return exports;
}));
