// mslpsapi/src/model/Xml2SomTemplate.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/InverseStore'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./InverseStore'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.Xml2SomTemplate = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.InverseStore);
  }
}(this, function(ApiClient, InverseStore) {
  'use strict';




  /**
   * The Xml2SomTemplate model module.
   * @module model/Xml2SomTemplate
   * @version v1
   */

  /**
   * Constructs a new <code>Xml2SomTemplate</code>.
   * @alias module:model/Xml2SomTemplate
   * @class
   */
  var exports = function() {
    var _this = this;





  };

  /**
   * Constructs a <code>Xml2SomTemplate</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Xml2SomTemplate} obj Optional instance to populate.
   * @return {module:model/Xml2SomTemplate} The populated <code>Xml2SomTemplate</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('DynamicForm')) {
        obj['DynamicForm'] = ApiClient.convertToType(data['DynamicForm'], 'Boolean');
      }
      if (data.hasOwnProperty('Order')) {
        obj['Order'] = ApiClient.convertToType(data['Order'], ['String']);
      }
      if (data.hasOwnProperty('Name2Node')) {
        obj['Name2Node'] = ApiClient.convertToType(data['Name2Node'], {'String': [Object]});
      }
      if (data.hasOwnProperty('InverseSearch')) {
        obj['InverseSearch'] = ApiClient.convertToType(data['InverseSearch'], {'String': InverseStore});
      }
    }
    return obj;
  }

  /**
   * @member {Boolean} DynamicForm
   */
  exports.prototype['DynamicForm'] = undefined;
  /**
   * @member {Array.<String>} Order
   */
  exports.prototype['Order'] = undefined;
  /**
   * @member {Object.<String, Array.<Object>>} Name2Node
   */
  exports.prototype['Name2Node'] = undefined;
  /**
   * @member {Object.<String, module:model/InverseStore>} InverseSearch
   */
  exports.prototype['InverseSearch'] = undefined;



  return exports;
}));
