// mslpsapi/src/model/XmpWriter.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/IXmpMeta'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./IXmpMeta'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.XmpWriter = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.IXmpMeta);
  }
}(this, function(ApiClient, IXmpMeta) {
  'use strict';




  /**
   * The XmpWriter model module.
   * @module model/XmpWriter
   * @version v1
   */

  /**
   * Constructs a new <code>XmpWriter</code>.
   * @alias module:model/XmpWriter
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>XmpWriter</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/XmpWriter} obj Optional instance to populate.
   * @return {module:model/XmpWriter} The populated <code>XmpWriter</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('XmpMeta')) {
        obj['XmpMeta'] = IXmpMeta.constructFromObject(data['XmpMeta']);
      }
      if (data.hasOwnProperty('ReadOnly')) {
        obj['ReadOnly'] = ApiClient.convertToType(data['ReadOnly'], 'Boolean');
      }
      if (data.hasOwnProperty('About')) {
        obj['About'] = ApiClient.convertToType(data['About'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {module:model/IXmpMeta} XmpMeta
   */
  exports.prototype['XmpMeta'] = undefined;
  /**
   * @member {Boolean} ReadOnly
   */
  exports.prototype['ReadOnly'] = undefined;
  /**
   * @member {String} About
   */
  exports.prototype['About'] = undefined;



  return exports;
}));
