// mslpsapi/src/api/LogsiticsCompanyApi.js
/**
 * MS.LPS.API
 * No descripton provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/BaseResponseResult'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/BaseResponseResult'));
  } else {
    // Browser globals (root is window)
    if (!root.Mslpsapi) {
      root.Mslpsapi = {};
    }
    root.Mslpsapi.LogsiticsCompanyApi = factory(root.Mslpsapi.ApiClient, root.Mslpsapi.BaseResponseResult);
  }
}(this, function(ApiClient, BaseResponseResult) {
  'use strict';

  /**
   * LogsiticsCompany service.
   * @module api/LogsiticsCompanyApi
   * @version v1
   */

  /**
   * Constructs a new LogsiticsCompanyApi. 
   * @alias module:api/LogsiticsCompanyApi
   * @class
   * @param {module:ApiClient} apiClient Optional API client implementation to use,
   * default to {@link module:ApiClient#instance} if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the logsiticsCompanyGetAllLogisticsCompanies operation.
     * @callback module:api/LogsiticsCompanyApi~logsiticsCompanyGetAllLogisticsCompaniesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/BaseResponseResult} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * 获取所有物流公司信息
     * @param {module:api/LogsiticsCompanyApi~logsiticsCompanyGetAllLogisticsCompaniesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/BaseResponseResult}
     */
    this.logsiticsCompanyGetAllLogisticsCompanies = function(callback) {
      var postBody = null;


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json', 'text/json'];
      var returnType = BaseResponseResult;

      return this.apiClient.callApi(
        '/api/LogsiticsCompany/GetAllLogisticsCompanies', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the logsiticsCompanyGetChannels operation.
     * @callback module:api/LogsiticsCompanyApi~logsiticsCompanyGetChannelsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/BaseResponseResult} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * 根据物流标识和物流公司配置信息获取渠道列表
     * @param {String} identity 物流标识
     * @param {Array.<String>} configs 物流公司配置信息
     * @param {module:api/LogsiticsCompanyApi~logsiticsCompanyGetChannelsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/BaseResponseResult}
     */
    this.logsiticsCompanyGetChannels = function(identity, configs, callback) {
      var postBody = null;

      // verify the required parameter 'identity' is set
      if (identity == undefined || identity == null) {
        throw "Missing the required parameter 'identity' when calling logsiticsCompanyGetChannels";
      }

      // verify the required parameter 'configs' is set
      if (configs == undefined || configs == null) {
        throw "Missing the required parameter 'configs' when calling logsiticsCompanyGetChannels";
      }


      var pathParams = {
      };
      var queryParams = {
        'identity': identity,
        'configs': this.apiClient.buildCollectionParam(configs, 'multi')
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json', 'text/json', 'application/x-www-form-urlencoded'];
      var accepts = ['application/json', 'text/json'];
      var returnType = BaseResponseResult;

      return this.apiClient.callApi(
        '/api/LogsiticsCompany/GetChannels', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the logsiticsCompanyGetConfigs operation.
     * @callback module:api/LogsiticsCompanyApi~logsiticsCompanyGetConfigsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/BaseResponseResult} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * 根据物流标识获取物流公司配置信息
     * @param {String} identity 物流标识
     * @param {module:api/LogsiticsCompanyApi~logsiticsCompanyGetConfigsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/BaseResponseResult}
     */
    this.logsiticsCompanyGetConfigs = function(identity, callback) {
      var postBody = null;

      // verify the required parameter 'identity' is set
      if (identity == undefined || identity == null) {
        throw "Missing the required parameter 'identity' when calling logsiticsCompanyGetConfigs";
      }


      var pathParams = {
      };
      var queryParams = {
        'identity': identity
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json', 'text/json'];
      var returnType = BaseResponseResult;

      return this.apiClient.callApi(
        '/api/LogsiticsCompany/GetConfigs', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));
